resolvers/node/resolution.rs

```diff
@@ -309,6 +321,20 @@ impl<
 
     let url_or_path =
       self.finalize_resolution(url, resolved_kind, Some(&referrer))?;
+    (|| {
+      let package_resolution_lookup_cache =
+        self.package_resolution_lookup_cache.as_ref()?;
+      if specifier_is_url
+        || specifier.starts_with("./")
+        || specifier.starts_with("../")
+        || specifier.starts_with("/")
+      {
+        return None;
+      }
+      let url = url_or_path.clone().into_url().ok()?;
+      package_resolution_lookup_cache.insert(url, specifier.to_string());
+      Some(())
+    })();
```

Comment:
Not too big of a fan of these IIFEs. I find it harder to follow along than if statements or extracting out a method/function.

Reply:
> I find it harder to follow along than if statements or extracting out a method/function.

I agree for reading but they're soooo easy to write 😄 Maybe at least naming the closure is more idiomatic.